src/onnx/parse_gemm.cpp

```diff
@@ -46,9 +46,19 @@ struct parse_gemm : op_parser<parse_gemm>
                            : args[0];
         auto l2 = (transb) ? info.add_instruction(make_op("transpose", {{"dims", perm}}), args[1])
                            : args[1];
+
+        if(alpha != 1.0f)
+        {
+            auto alpha_literal   = info.add_literal(alpha);
+            auto alpha_broadcast = info.add_instruction(
+                make_op("multibroadcast", {{"output_lens", l1->get_shape().lens()}}),
+                alpha_literal);
+            l1 = info.add_instruction(make_op("mul"), l1, alpha_broadcast);
+        }
```

Comment:
The alpha_broadcast can be of different data type from l1, so we need a check to see if l1 and alpha_broadcast have the same data type. If not, need to add a convert operator. But there is another problem if we do the data type conversion.

According to the ONNX spec, alpha a float, and l1 can be int32, for this case, we need to convert alpha(float) to a int32, which will lose data precision. For example, 0.1 will be converted to 0. 

The same is for the beta.

Reply:
@pfultz2 any opinion on this?

Reply:
We can convert it to float and do the multiply and then convert back. I think we should just move this common type logic to `add_broadcastable_binary_op`, which should simplify this to:

```cpp
auto multiply = info.add_broadcastable_binary_op("mul", l1, alpha_literal);
l1 = info.add_instruction(make_op("convert", {{"target_type", l1->get_shape().type()}}), multiply);
```

Reply:
So PR #875 adds such support. So once its merged then `add_broadcastable_binary_op` can be used here.

Reply:
This type conversion is fine. The main problem is that, for example, if l1 is int32, then alpha will be converted from float to int32. This convert will lose information. 
One example is: alpha is 0.5(float), after the type conversion, alpha is converted to 0(int32), which is totally wrong.

Reply:
> if l1 is int32, then alpha will be converted from float to int32.

With PR #875, it wont convert `alpha` to int32. Instead, `l1` is converted to float, which is why we need this conversion back to int32.

Reply:
Made changes to make use of `broadcastable_binary_op`